coroutines/src/commonMain/kotlin/SynchronizedMap.kt

```diff
@@ -0,0 +1,115 @@
+package com.juul.tuulbox.coroutines
```

Comment:
This doesn't seem like it belongs in `coroutines` package/module.
Hate to add the burden of adding another module, but seems like it would be better to be given a home in `collections` module?